arangod/Agency/Supervision.cpp

```diff
@@ -1010,6 +1011,101 @@ void Supervision::workJobs() {
+              if (planIndex.hasKey("isBuilding") && collection.has("shards")) {
+                auto const& planId = planIndex.get("id");
+                std::unordered_set<std::string> curServers;
+                auto const& shards = collection("shards");
+                auto const& nshards = collection.hasAsUInt("numberOfShards");
+                if (!nshards.second) {
+                  continue;
+                }
+                size_t nIndexes = 0;
+                for (auto const& sh : shards.children()) {
+
+                  auto const& shname = sh.first;
+                  
+                  if (currentDBs.has(colPath + shname + "/indexes")) {
+                    auto const& curIndexes =
+                      currentDBs(colPath + shname + "/indexes").slice();
+                    for (auto const& curIndex : VPackArrayIterator(curIndexes)) {
+                      auto const& curId = curIndex.get("id");
+                      if (planId == curId) {
+                        ++nIndexes;
+                      }
+                    }
+                  }
+                }
+                if (nIndexes == nshards.first) {
+                  built.emplace(planId.copyString());
+                }
+              }
+            }
+          }
+
+          // We have some indexes, that have been built and are isBuilding still
+          if (!built.empty()) {
+
+            auto envelope = std::make_shared<Builder>();
+            { VPackArrayBuilder trxs(envelope.get());
+              { VPackArrayBuilder trx(envelope.get());
+                { VPackObjectBuilder operation(envelope.get());
+                  envelope->add(
+                    VPackValue(
+                      _agencyPrefix + planColPrefix + colPath + "indexes"));
+                  VPackArrayBuilder value(envelope.get());
+                  for (auto const& planIndex :
+                         VPackArrayIterator(indexes.first)) {
+                    if (built.find(
+                          planIndex.get("id").copyString()) != built.end()) {
+                      { VPackObjectBuilder props(envelope.get());
+                        for (auto const& prop : VPackObjectIterator(planIndex)) {
+                          auto const& key = prop.key.copyString();
+                          if (key != "isBuilding") {
+                            envelope->add(key, prop.value);
+                          }
+                        }}
+                    } else {
+                      envelope->add(planIndex);
+                    }
+                  }
+                }
+                { VPackObjectBuilder precondition(envelope.get());
+                  envelope->add(
+                    VPackValue(
+                      _agencyPrefix + planColPrefix + colPath + "indexes")); 
+                  envelope->add(indexes.first); }
+              }}
+            write_ret_t res = _agent->write(envelope);
```

Comment:
Do you increase the plan version? This should be important, otherwise db servers my disagree on wether this index can be used or not.

Reply:
good point